quantum/joystick.h

```diff
@@ -8,9 +12,13 @@
 #    define JOYSTICK_AXES_COUNT 4
 #endif
 
-#include "quantum.h"
+#ifndef JOYSTICK_AXES_RESOLUTION
+#    define JOYSTICK_AXES_RESOLUTION 8
+#elif JOYSTICK_AXES_RESOLUTION < 8 || JOYSTICK_AXES_RESOLUTION > 16
+#    error JOYSTICK_AXES_RESOLUTION must be between 8 and 16
+#endif
 
-#include <stdint.h>
+#define JOYSTICK_RESOLUTION ((1L << (JOYSTICK_AXES_RESOLUTION - 1)) - 1)
```

Comment:
```suggestion
#define JOYSTICK_RESOLUTION ((1U << (JOYSTICK_AXES_RESOLUTION - 1)) - 1)
```

Reply:
It's actually `1L` that works, not `1U`, sorry for the confusion. `1U` behaves the same as `1UL`, i.e. it doesn't produce any joystick output. The variable needs to be signed for the code to work.

Reply:
Huh, alright then.